sys/net/gnrc/network_layer/ipv6/netif/gnrc_ipv6_netif.c

```diff
@@ -123,6 +123,15 @@ static ipv6_addr_t *_add_addr_to_entry(gnrc_ipv6_netif_t *entry, const ipv6_addr
                 DEBUG("ipv6_netif: error adding prefix to 6LoWPAN-ND management\n");
             }
 #endif
+#ifdef MODULE_GNRC_SIXLOWPAN_ND_ROUTER
+            if ((entry->flags & GNRC_IPV6_NETIF_FLAGS_ROUTER) &&
```

Comment:
At least the checks for ROUTER and RTR_ADV (and even the the mutex_lock()) you could combine with the above send-out